src/Migrations/M032MigrateCampaignCoversToP4ColumnsBlock.php

```diff
@@ -0,0 +1,203 @@
+            if (empty($post->post_content)) {
+                continue;
+            }
+
+            echo 'Parsing post ', $post->ID, "\n"; // phpcs:ignore
+            $result = false;
+
+            // Go through the post blocks to find the planet4-blocks/covers one.
+            $blocks = $parser->parse($post->post_content);
+
+            foreach ($blocks as &$block) {
+                // For older cover blocks, the cover type is empty, so use the default content cover type,hence skip it.
+                if (!isset($block['attrs']['cover_type'])) {
+                    continue;
+                }
+
+                // For old cover type(earlier it was numeric).
+                if (is_numeric($block['attrs']['cover_type'])) {
+                    $block['attrs']['cover_type'] = self::OLD_COVER_TYPES[ $block['attrs']['cover_type'] ];
+                }
+
+                // Skip other blocks & non campaign cover blocks.
+                if (
+                    ! isset($block['blockName']) || $block['blockName'] !== self::BLOCK_NAME ||
+                    $block['attrs']['cover_type'] !== self::COVER_TYPE
+                ) {
+                    continue;
+                }
+
+                // Gathering the data we want from this block.
+                $block_attrs = self::get_columns_block_attrs($block);
+
+                // No data, skip this block.
+                if (empty($block_attrs)) {
+                    continue;
+                }
+
+                $block_attrs = array_merge(['columns_block_style' => 'image'], $block_attrs);
+                $block_attrs['className'] = 'is-style-image';
+                $block = self::transform_block($block_attrs);
+            }
+
+            // Unset the reference to prevent potential issues.
+            unset($block);
+
+            // Serialize the blocks content.
+            $new_content = serialize_blocks($blocks);
+
+            if ($post->post_content !== $new_content) {
+                $post_update = array(
+                    'ID' => $post->ID,
+                    'post_content' => $new_content,
+                );
+
+                try {
+                    // Update the post with the replaced blocks.
+                    wp_update_post($post_update);
+                    $result = true;
+                } catch (\Throwable $e) {
+                    echo 'Error on post ', $post->ID, "\n";
+                    echo $e->getMessage(), "\n";
+                }
+            }
+
+            echo $result
+                ? "Migration successful\n"
+                : "Migration wasn't executed\n"; // phpcs:ignore
+        }
+    }
+    // phpcs:enable SlevomatCodingStandard.Functions.UnusedParameter
+
+    /**
+     * Get all the posts using block name.
+     *
+     * @param string $block_name - A block name.
+     * @return mixed - The posts using block or null if no posts are found.
+     */
+    private static function get_posts_using_block(string $block_name): mixed
```

Comment:
This function is also used [here](https://github.com/greenpeace/planet4-master-theme/pull/2363/files#diff-882ae83a59536852c35295e4efeeaf80e277d1f512cb5c10d2143e9fbad46c76R88) and [here](https://github.com/greenpeace/planet4-master-theme/pull/2357/files#diff-053e954328867930e539fca66766722a453a9f544400e19f85cc1bacb0b2680fR110).
Do you think it is worth moving it to a separate file to avoid duplicate content?